libraries/Microsoft.Bot.Builder/RootFrame.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder
+{
+    public class RootFrame : IFrame
+    {
+        private readonly string _cacheKey = $"RootFrame-{Guid.NewGuid()}";
```

Comment:
I think this is a good default, but I believe a constructor overload that allows people to supply their own name for this would be useful for logging/debugging scenarios.